File: old2/org.eclipse.cdt.windows.debug.native/src/HRESULTFailure.cpp
/**********************************************************************
 * Copyright (c) 2006 QNX Software Systems and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors: 
 *     QNX Software Systems - Initial API and implementation
 **********************************************************************/
#include <windows.h>
#include <jni.h>
#include <comdef.h>

#include "HRESULTFailure.h"

void throwHRESULT(JNIEnv * env, HRESULT hr, char * file, int line) {
	jclass cls = env->FindClass("org/eclipse/cdt/windows/debug/core/HRESULTFailure");
	if (cls == NULL)
		return;
	
	jmethodID constructor = env->GetMethodID(cls, "<init>", "(ILjava/lang/String;Ljava/lang/String;I)V");
	if (constructor == NULL)
		return;

	_com_error error(hr);
	jstring message = env->NewStringUTF(error.ErrorMessage());
	jstring _file = env->NewStringUTF(file);
	
	env->Throw((jthrowable)env->NewObject(cls, constructor, (jint)hr, message, _file, line)); 
}

File: windows/old/really_old/org.eclipse.cdt.windows.debug.core/src/org/eclipse/cdt/windows/debug/core/HRESULTFailure.java
/**********************************************************************
 * Copyright (c) 2006 QNX Software Systems and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors: 
 *     QNX Software Systems - Initial API and implementation
 **********************************************************************/

package org.eclipse.cdt.windows.debug.core;

/**
 * @author Doug Schaefer
 *
 * This exception holds the HRESULT and is thrown when a Windows operation fails.
 */
public class HRESULTFailure extends Exception {

	private static final long serialVersionUID = 1;
	
	@SuppressWarnings("unused")
	private int hr;
	@SuppressWarnings("unused")
	private String file;
	@SuppressWarnings("unused")
	private int line;
	
	@SuppressWarnings("unused")
	private HRESULTFailure(int hr, String message, String file, int line) {
		super(message);
		this.hr = hr;
		this.file = file;
		this.line = line;
	}
	
	public static final int S_OK = 0;
	public static final int S_FALSE = 1;
	public static final int E_UNEXPECTED = 0x8000FFFF;
	
	public int getHR() {
		return hr;
	}
	
}
